components/locid/src/langid.rs

```diff
@@ -142,6 +143,39 @@ impl LanguageIdentifier {
         let lang_id = Self::from_bytes(input.as_ref())?;
         Ok(lang_id.to_string())
     }
+
+    /// Compare this `LanguageIdentifier` with a BCP-47 string.
+    ///
+    /// The return value is equivalent to what would happen if you first converted this
+    /// `LanguageIdentifier` to a BCP-47 string and then performed a byte comparison.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locid::LanguageIdentifier;
+    /// use std::cmp::Ordering;
+    ///
+    /// let bcp47_strings: &[&[u8]] = &[
+    ///     b"pl-Latn-PL",
+    ///     b"und",
+    ///     b"und-Adlm",
+    ///     b"und-GB",
+    ///     b"und-ZA",
+    ///     b"und-fonipa",
+    ///     b"zh",
+    /// ];
+    ///
+    /// for ab in bcp47_strings.windows(2) {
+    ///     let a = ab[0];
+    ///     let b = ab[1];
+    ///     assert!(a.cmp(b) == Ordering::Less);
+    ///     let a_langid = LanguageIdentifier::from_bytes(a).expect(&String::from_utf8_lossy(a));
+    ///     assert!(a_langid.cmp_bytes(b) == Ordering::Less);
+    /// }
+    /// ```
```

Comment:
`nit`: can you document in the comment that this is a strict case-sensitive, canonicalized comparison?
The whole Locale/LangId comparison to string is tricky because it can be "matching" on several levels of canonicalization and strictness. In this case, we will compare strictly without any normalization/canonicalization of the `other`.